### ISSUE TITLE
[backend/core] User.applications should be optional

### ISSUE THREAD

**Issue by bk3c** [bloom-housing/bloom#407]
Flagged this because of a jest failure (yay tests!):

```
  ● Test suite failed to run

    __tests__/authentication/RequireLogin.test.tsx:17:7 - error TS2741: Property 'applications' is missing in type '{ id: string; email: string; passwordHash: string; firstName: string; lastName: string; dob: Date; createdAt: Date; updatedAt: Date; }' but required in type 'User'.

    17 const mockUser: User = {
             ~~~~~~~~

      ../../backend/core/src/entity/user.entity.ts:35:3
        35   applications: Application[];
             ~~~~~~~~~~~~
        'applications' is declared here.
```

I think that we just need a question mark after applications in the definition file, since many users won't have applications (and none will have applications when the user is first created). If I add one, then the tests pass again, but I just want to make sure that's the right thing for `backend/core` as a whole.

**Comment by pbn4** [comment]
@bk3c I'm sorry for breaking the tests. :)  TypeORM will return an empty Array when user has no applications so I believe we should fix tests instead https://github.com/bloom-housing/bloom/pull/408.

**Comment by bk3c** [comment]
Fixed in #408
